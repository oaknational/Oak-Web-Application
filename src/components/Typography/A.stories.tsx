import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import Component from "./A";

export default {
  title: "Typography/A",
  component: Component,
} as ComponentMeta<typeof Component>;

const Template: ComponentStory<typeof Component> = (args) => (
  <Component {...args}>Click me!.</Component>
);

export const A = Template.bind({});
A.args = {
  fontWeight: 600,
  color: "inYourFace",
  href: "/jkj",
};
