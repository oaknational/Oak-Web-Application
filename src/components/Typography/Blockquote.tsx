import styled from "styled-components";

import { box, BoxProps } from "../Box";

export const Blockquote = styled.blockquote<BoxProps>`
  ${box}
`;

export default Blockquote;
