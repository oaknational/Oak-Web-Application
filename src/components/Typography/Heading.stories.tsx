import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Heading as Component } from ".";

export default {
  title: "Typography/Heading",
  component: Component,
  argTypes: {
    tag: {
      defaultValue: "h1",
    },
  },
} as ComponentMeta<typeof Component>;

const Template: ComponentStory<typeof Component> = (args) => (
  <Component {...args}>Heading 1</Component>
);

export const Heading = Template.bind({});
Heading.args = {
  tag: "h1",
};
