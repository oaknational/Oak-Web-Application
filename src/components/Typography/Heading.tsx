import { FC } from "react";
import styled, { css } from "styled-components";

import color from "../../styles/utils/color";
import { ResponsiveValues } from "../../styles/utils/responsive";
import { margin, MarginProps } from "../../styles/utils/spacing";
import typography, { TypographyProps } from "../../styles/utils/typography";

export const HEADING_TAGS = ["h1", "h2", "h3", "h4", "h5", "h6"] as const;

export type HeadingTag = typeof HEADING_TAGS[number];

export type HeadingTagProps = {
  tag: HeadingTag;
};
const HeadingTagComponent: FC<HeadingTagProps> = (props) => {
  const { tag, ...otherProps } = props;
  const Tag = tag;
  return <Tag {...otherProps} />;
};

const headingSizes = {
  56: "56px",
  48: "48px",
  40: "40px",
  32: "32px",
  24: "24px",
  20: "20px",
  16: "16px",
};
export type HeadingFontSize = keyof typeof headingSizes;
export const headingDefaults = css`
  font-weight: 600;
  font-family: Lexend, sans-serif;
  line-height: 1.2;
`;

type HeadingProps = Omit<TypographyProps, "fontSize"> &
  HeadingTagProps & {
    fontSize: ResponsiveValues<HeadingFontSize>;
  } & MarginProps;

const Heading = styled(HeadingTagComponent)<HeadingProps>`
  ${headingDefaults}
  ${margin}
  ${typography}
  ${color}
`;

export default Heading;
