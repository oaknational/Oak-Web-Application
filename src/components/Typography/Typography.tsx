import React, { FC } from "react";
import clsx from "clsx";

import styles from "../../styles/Typography.module.css";

// Defining the HTML tag that the component will support
type VariantsType =
  | "h1"
  | "h2"
  | "h3"
  | "h4"
  | "h5"
  | "h6"
  | "body1"
  | "body2"
  | "body3"
  | "body4";

const variantsMapping: Record<VariantsType, keyof JSX.IntrinsicElements> = {
  h1: "h1",
  h2: "h2",
  h3: "h3",
  h4: "h4",
  h5: "h5",
  h6: "h6",
  body1: "p",
  body2: "p",
  body3: "p",
  body4: "p",
};

type TypographyProps = {
  semanticVariant: VariantsType;
  styleVariant?: VariantsType;
  className?: string;
};

const Text: FC<TypographyProps> = ({
  semanticVariant,
  styleVariant,
  children,
  ...props
}) => {
  const Component = variantsMapping[semanticVariant];
  const cssVariant = styleVariant || semanticVariant;
  return (
    <Component
      className={clsx(styles[`typography--variant-${cssVariant}`])}
      {...props}
    >
      {children}
    </Component>
  );
};

export default Text;
