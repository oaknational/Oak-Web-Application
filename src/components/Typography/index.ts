export { default as Heading } from "./Heading";
export type { HeadingTag } from "./Heading";
export { default } from "./Typography";
export { default as Span } from "./Span";
export { default as P } from "./P";
export { default as Hr } from "./Hr";
export { default as OL } from "./OL";
export { default as LI } from "./LI";
export { default as UL } from "./UL";
