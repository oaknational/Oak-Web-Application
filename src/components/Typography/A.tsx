import Link, { LinkProps } from "next/link";
import styled, { CSSProperties } from "styled-components";

import { margin, MarginProps } from "../../styles/utils/spacing";
import typography, { TypographyProps } from "../../styles/utils/typography";

import { BodyFontSize } from "./Typography";

type AProps = MarginProps &
  LinkProps &
  Omit<TypographyProps, "fontSize" | "fontFamily"> & {
    fontSize?: BodyFontSize;
    textAlign?: CSSProperties["textAlign"];
  };
/**
 * Styled `a` (next/link) (anchor) component.
 * ##
 */

const A = styled(Link)<AProps>`
  ${typography}
  ${margin};
  text-align: ${(props) => props.textAlign};
`;

export default A;
