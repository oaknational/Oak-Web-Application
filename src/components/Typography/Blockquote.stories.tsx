import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Blockquote as Component } from "./Blockquote";

export default {
  title: "Typography/Blockquote",
  component: Component,
  argTypes: {},
} as ComponentMeta<typeof Component>;

const Template: ComponentStory<typeof Component> = (args) => (
  <Component {...args} cite="https://www.huxley.net/bnw/four.html">
    &ldquo; Words can be like X-rays, if you use them properly—they’ll go
    through anything. You read and you’re pierced. &rdquo;
  </Component>
);

export const Blockquote = Template.bind({});
