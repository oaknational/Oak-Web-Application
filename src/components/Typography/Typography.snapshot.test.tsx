import { render } from "@testing-library/react";

import Typography from "./Typography";

it("renders <Typography> unchanged", () => {
  const { container } = render(
    <Typography semanticVariant="h2">lesson content</Typography>
  );
  expect(container).toMatchSnapshot();
});
