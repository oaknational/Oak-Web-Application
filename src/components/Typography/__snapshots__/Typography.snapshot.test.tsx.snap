// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Typography> unchanged 1`] = `
<div>
  <h2
    class="typography--variant-h2"
  >
    lesson content
  </h2>
</div>
`;
